test/sslapitest.c

```diff
@@ -603,42 +489,34 @@ static int execute_test_large_message(const SSL_METHOD *smeth,
      * We assume the supplied certificate is big enough so that if we add
      * NUM_EXTRA_CERTS it will make the overall message large enough. The
      * default buffer size is requested to be 16k, but due to the way BUF_MEM
-     * works, it ends up allocating a little over 21k (16 * 4/3). So, in this test
-     * we need to have a message larger than that.
+     * works, it ends up allocating a little over 21k (16 * 4/3). So, in this
+     * test we need to have a message larger than that.
      */
     certlen = i2d_X509(chaincert, NULL);
-    OPENSSL_assert((certlen * NUM_EXTRA_CERTS)
-                   > ((SSL3_RT_MAX_PLAIN_LENGTH * 4) / 3));
+    if (!TEST_long_gt(certlen * NUM_EXTRA_CERTS,
+                      (SSL3_RT_MAX_PLAIN_LENGTH * 4) / 3))
+        goto end;
```

Comment:
I would keep the assertion, 'cause if I understand this correctly, if that particular condition fails, something is wrong with the test program itself.